source/dotnet/Library/AdaptiveCards.Rendering.Wpf/AdaptiveRichTextBlockRenderer.cs

```diff
@@ -0,0 +1,200 @@
+                    Mode = BindingMode.OneWay,
+                    Converter = new MultiplyConverter(richTB.MaxLines)
+                });
+
+                // Add both to a grid so they go as a unit
+                uiGrid.Children.Add(measureBlock);
+
+                uiGrid.Children.Add(uiRichTB);
+                return uiGrid;
+
+            }
+
+            if(!richTB.IsVisible)
+            {
+                uiRichTB.Visibility = Visibility.Collapsed;
+            }
+
+            return uiRichTB;
+        }
+
+        private static Inline FormatInlineTextRun(AdaptiveRichTextBlock.AdaptiveParagraph.AdaptiveTextRun textRun, AdaptiveRenderContext context)
+        {
+            Marked marked = new Marked();
+            marked.Options.Renderer = new AdaptiveXamlMarkdownRenderer();
+            marked.Options.Mangle = false;
+            marked.Options.Sanitize = true;
+
+            // Handle Date/Time parsing
+            string text = RendererUtilities.ApplyTextFunctions(textRun.Text, context.Lang);
+
+            // Handle markdown
+            string xaml = $"<Span  xmlns=\"http://schemas.microsoft.com/winfx/2006/xaml/presentation\">{marked.Parse(text)}</Span>";
+            StringReader stringReader = new StringReader(xaml);
+            XmlReader xmlReader = XmlReader.Create(stringReader);
+
+            Span uiInlineElement = (Span)XamlReader.Load(xmlReader);
+            uiInlineElement.Style = context.GetStyle($"Adaptive.{textRun.Type}");
+
+            uiInlineElement.FontFamily = new FontFamily(context.Config.GetFontFamily(textRun.FontStyle));
+            uiInlineElement.FontWeight = FontWeight.FromOpenTypeWeight(context.Config.GetFontWeight(textRun.FontStyle, textRun.Weight));
+            uiInlineElement.FontSize = context.Config.GetFontSize(textRun.FontStyle, textRun.Size);
+
+            FontColorConfig colorOption;
+            switch (textRun.Color)
+            {
+                case AdaptiveTextColor.Accent:
+                    colorOption = context.ForegroundColors.Accent;
+                    break;
+                case AdaptiveTextColor.Attention:
+                    colorOption = context.ForegroundColors.Attention;
+                    break;
+                case AdaptiveTextColor.Dark:
+                    colorOption = context.ForegroundColors.Dark;
+                    break;
+                case AdaptiveTextColor.Good:
+                    colorOption = context.ForegroundColors.Good;
+                    break;
+                case AdaptiveTextColor.Light:
+                    colorOption = context.ForegroundColors.Light;
+                    break;
+                case AdaptiveTextColor.Warning:
+                    colorOption = context.ForegroundColors.Warning;
+                    break;
+                case AdaptiveTextColor.Default:
+                default:
+                    colorOption = context.ForegroundColors.Default;
+                    break;
+            }
```

Comment:
is there not already a helper for this?

Reply:
Huh. Overlooked making a helper for WPF. Just added a helper for it.

---
In reply to: [263170355](https://github.com/Microsoft/AdaptiveCards/pull/2519#discussion_r263170355) [](ancestors = 263170355)